url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/org/jgraph/graph/PortRenderer.java

```diff
@@ -0,0 +1,212 @@
+		setForeground(UIManager.getColor("MenuItem.selectionBackground"));
+		setBackground(UIManager.getColor("Tree.selectionBorderColor"));
+	}
+
+	/**
+	 * Configure and return the renderer based on the passed in components. The
+	 * value is typically set from messaging the graph with
+	 * <code>convertValueToString</code>.
+	 * 
+	 * @param graph
+	 *            the graph that that defines the rendering context.
+	 * @param view
+	 *            the cell view that should be rendered.
+	 * @param sel
+	 *            whether the object is selected.
+	 * @param focus
+	 *            whether the object has the focus.
+	 * @param preview
+	 *            whether we are drawing a preview.
+	 * @return the component used to render the value.
+	 */
+	public Component getRendererComponent(JGraph graph, CellView view,
+			boolean sel, boolean focus, boolean preview) {
+		// Check type
+		if (view instanceof PortView && graph != null) {
+			graphBackground = graph.getBackground();
+			this.view = (PortView) view;
+			this.hasFocus = focus;
+			this.selected = sel;
+			this.preview = preview;
+			this.xorEnabled = graph.isXorEnabled();
+			return this;
+		}
+		return null;
+	}
+
+	/**
+	 * Paint the renderer. Overrides superclass paint to add specific painting.
+	 * Note: The preview flag is interpreted as "highlight" in this context.
+	 * (This is used to highlight the port if the mouse is over it.)
+	 */
+	public void paint(Graphics g) {
+		Dimension d = getSize();
+		if (xorEnabled) {
+			g.setColor(graphBackground);
+			g.setXORMode(graphBackground);
+		}
+		super.paint(g);
+		if (preview) {
+			g.fill3DRect(0, 0, d.width, d.height, true);
+		} else {
+			g.fillRect(0, 0, d.width, d.height);
+		}
+		boolean offset = (GraphConstants.getOffset(view.getAllAttributes()) != null);
+		g.setColor(getForeground());
+		if (!offset)
+			g.fillRect(1, 1, d.width - 2, d.height - 2);
+		else if (!preview)
+			g.drawRect(1, 1, d.width - 3, d.height - 3);
+	}
+
+	/**
+	 * Overridden for performance reasons. See the <a
+	 * href="#override">Implementation Note</a> for more information.
+	 */
+	public void validate() {
+	}
+
+	/**
+	 * Overridden for performance reasons. See the <a
+	 * href="#override">Implementation Note</a> for more information.
+	 */
+	public void revalidate() {
+	}
+
+	/**
+	 * Overridden for performance reasons. See the <a
+	 * href="#override">Implementation Note</a> for more information.
+	 */
+	public void repaint(long tm, int x, int y, int width, int height) {
+	}
+
+	/**
+	 * Overridden for performance reasons. See the <a
+	 * href="#override">Implementation Note</a> for more information.
+	 */
+	public void repaint(Rectangle r) {
+	}
+
+	/**
+	 * Overridden for performance reasons. See the <a
+	 * href="#override">Implementation Note</a> for more information.
+	 */
+	protected void firePropertyChange(String propertyName, Object oldValue,
+			Object newValue) {
+		// Strings get interned...
+		if (propertyName == "text")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.